packages/generator/src/edmx-parser/edmx-file-reader.ts

```diff
@@ -1,22 +1,112 @@
 import { PathLike, readFileSync } from 'fs';
 import path, { basename } from 'path';
 import { parse } from 'fast-xml-parser';
-import { createLogger, ODataVersion } from '@sap-cloud-sdk/util';
+import { assoc, createLogger, flat, ODataVersion } from '@sap-cloud-sdk/util';
 import { forceArray } from '../generator-utils';
 import { SwaggerMetadata } from '../swagger-parser/swagger-types';
+import {
```

Comment:
Some time ago I refactored the generator https://github.com/SAP/cloud-sdk-js/pull/257

One part was to get rid of the EDM types and just pass the root down to the VDM methods and do the parsing when it is needed. At this point you know in general if you are V2 or V4 and can make some assumptions on the types. I guess the new types are just super unions containing all possible things. But then the question is how usefull they are.

Reply:
The problem, I am not able to understand yet if the logiv from line 170 onwards needs to be done here or if you could do it also on the fly in the parse methods.

Reply:
I think I have to understand firt (lets discuss tomorrow) How stront the schema separates the types and so on. Is it like a virtual to service definitions? If so one could also pull the changes up in the generator loop for the services and generate multiple services for each namespace?